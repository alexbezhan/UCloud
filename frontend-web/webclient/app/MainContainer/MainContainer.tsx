import {ReduxObject, ResponsiveReduxObject} from "DefaultObjects";
import {LoadableContent} from "LoadableContent";
import Spinner from "LoadingIcon/LoadingIcon";
import * as React from "react";
import {connect} from "react-redux";
import styled from "styled-components";
import {Absolute, Box, Hide} from "ui-components";
import * as Heading from "ui-components/Heading";

export interface MainContainerStateProps {
    responsiveState?: ResponsiveReduxObject;
}

export interface MainContainerProps extends MainContainerStateProps {
    sidebar?: React.ReactNode;
    sidebarSize?: number;
    main: React.ReactNode;
    additional?: React.ReactNode;
    header?: React.ReactNode;
    headerSize?: number;
}


// eslint-disable-next-line no-underscore-dangle
export const _MainContainer = ({
    sidebar,
    main,
    additional,
    header,
    sidebarSize = 240,
    headerSize = 96,
    responsiveState
}: MainContainerProps): JSX.Element => {
    const leftSidebarSize = responsiveState!.greaterThan.xl ? 190 : 68; // main website sidebar H size
    const topMenuSize = 48; // main website top menu V size
    const pad = 14; // padding unit

    const mainYpad = header ? headerSize : pad;
    const mainXpad = sidebar && responsiveState!.greaterThan.md ? sidebarSize : pad;


    return (
        <React.StrictMode>
            <Box backgroundColor="white" ml={leftSidebarSize} pt={topMenuSize} pb={pad} pl={pad} pr="0">
                {header && (
                    <HeaderContainer
                        top={topMenuSize}
                        left="0"
                        py={pad}
                        pl={leftSidebarSize + pad}
                        pr={pad}
                        width={1}
                        height={headerSize}
                        bg="white"
                    >
                        {header}
                    </HeaderContainer>
                )}
                {sidebar && (
                    <Hide sm xs md>
                        <SidebarContainer
                            height="100%"
                            pt={topMenuSize + mainYpad}
                            top="0"
                            right="0"
                            px={pad}
                            width={sidebarSize}
                        >
                            {sidebar}
                        </SidebarContainer>
                    </Hide>
                )}
                <Box pt={mainYpad} pr={mainXpad}>
                    <Hide lg xl xxl>
                        {sidebar}
                    </Hide>
                    {main}
                </Box>
                {additional}
            </Box>
        </React.StrictMode>
    );
};

const mapStateToProps = ({responsive}: ReduxObject): MainContainerStateProps => ({
    responsiveState: responsive
});

export const MainContainer = connect<MainContainerStateProps>(mapStateToProps)(_MainContainer);

export interface LoadingMainContainerProps extends MainContainerProps {
    loading?: boolean;
    error?: string;
    fallbackHeader?: JSX.Element;
    fallbackSidebar?: JSX.Element;
}

export const LoadingMainContainer: React.FunctionComponent<LoadingMainContainerProps> = props => {
    let main: React.ReactNode;
    if (props.loading) {
        main = <Spinner />;
    } else if (props.error !== undefined) {
        main = <Heading.h2>{props.error}</Heading.h2>;
    } else {
        main = props.main;
    }

    const header = props.loading || props.error !== undefined ? props.fallbackHeader : props.header;
    const sidebar = props.loading || props.error !== undefined ? props.fallbackSidebar : props.sidebar;

    return (
        <MainContainer
            {...props}
            main={main}
            header={header}
            sidebar={sidebar}
        />
    );
};

export interface LoadableMainContainerProps<T = any> extends MainContainerProps {
    loadable: LoadableContent<T>;
    fallbackHeader?: JSX.Element;
    fallbackSidebar?: JSX.Element;
}

export function LoadableMainContainer(props: LoadableMainContainerProps): JSX.Element {
    if (!props.loadable.content) {
        const main = !!props.loadable.error ?
            <Heading.h2>{props.loadable.error.statusCode} - {props.loadable.error.errorMessage}</Heading.h2> :
            <Spinner />;
        return (
            <MainContainer
                header={props.fallbackHeader}
                sidebar={props.fallbackSidebar}
                main={main}
            />
        );
    } else {
        return <MainContainer {...props} />;
    }
}


const HeaderContainer = styled(Absolute)`
    position: fixed;
    z-index: 50;
`;

const SidebarContainer = styled(Absolute)`
    overflow-y: auto;
    position: fixed;
    z-index: 45;
`;
