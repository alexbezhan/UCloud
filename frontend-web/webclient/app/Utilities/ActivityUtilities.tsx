import { buildQueryString } from "./URIUtilities";
import { ScrollRequest } from "Scroll";
import { ActivityFilter } from "Activity";

export const activityQuery = (scroll: ScrollRequest<number>, filter?: ActivityFilter) => {
    let params: any = {};
    if (!!scroll.offset) params.offset = scroll.offset;
    params.scrollSize = scroll.scrollSize;

    if (filter !== undefined) {
        if (filter.type !== undefined) params.type = filter.type;
        if (filter.collapseAt !== undefined) params.collapseAt = filter.collapseAt;
        if (filter.maxTimestamp !== undefined) params.maxTimestamp = filter.maxTimestamp.getTime();
        if (filter.minTimestamp !== undefined) params.minTimestamp = filter.minTimestamp.getTime();
    }

    return buildQueryString("/activity/browse/user", params);
};

export const activityStreamByPath = (path: string) =>
    `/activity/by-path?path=${encodeURIComponent(path)}`;

export const getLatestActivity = () => undefined;