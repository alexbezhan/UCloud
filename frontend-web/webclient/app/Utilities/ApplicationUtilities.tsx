import {
    ApplicationInvocationDescription,
    ApplicationMetadata,
    ApplicationParameter,
    AppState,
    ParameterTypes,
    RunsSortBy
} from "Applications";
import Cloud from "Authentication/lib";
import {SortOrder} from "Files";
import * as React from "react";
import {snackbarStore} from "Snackbar/SnackbarStore";
import {Page} from "Types";
import {addStandardDialog} from "UtilityComponents";
import {errorMessageOrDefault, removeTrailingSlash} from "UtilityFunctions";
import {expandHomeFolder} from "./FileUtilities";

export const hpcJobQueryPost = "/hpc/jobs";

export const hpcJobQuery = (
    id: string,
    stdoutLine: number,
    stderrLine: number,
    stdoutMaxLines: number = 1000,
    stderrMaxLines: number = 1000
) =>
    `/hpc/jobs/follow/${encodeURIComponent(id)}?stdoutLineStart=${stdoutLine}&stdoutMaxLines=${stdoutMaxLines}&stderrLineStart=${stderrLine}&stderrMaxLines=${stderrMaxLines}`;

export function hpcJobsQuery(
    itemsPerPage: number,
    page: number,
    sortOrder?: SortOrder,
    sortBy?: RunsSortBy,
    minTimestamp?: number,
    maxTimestamp?: number,
    filter?: AppState
): string {
    let query = `/hpc/jobs/?itemsPerPage=${itemsPerPage}&page=${page}`;
    if (sortOrder) query = query.concat(`&order=${sortOrder}`);
    if (sortBy) query = query.concat(`&sortBy=${sortBy}`);
    if (minTimestamp != null) query = query.concat(`&minTimestamp=${minTimestamp}`);
    if (maxTimestamp != null) query = query.concat(`&maxTimestamp=${maxTimestamp}`);
    if (filter != null) query = query.concat(`&filter=${filter}`);
    return query;
}

export const hpcFavoriteApp = (name: string, version: string) =>
    `/hpc/apps/favorites/${encodeURIComponent(name)}/${encodeURIComponent(version)}`;

export const hpcFavorites = (itemsPerPage: number, pageNumber: number) =>
    `/hpc/apps/favorites?itemsPerPage=${itemsPerPage}&page=${pageNumber}`;

export const hpcApplicationsQuery = (page: number, itemsPerPage: number) =>
    `/hpc/apps?page=${page}&itemsPerPage=${itemsPerPage}`;

interface HPCApplicationsSearchQuery {
    query: string,
    page: number,
    itemsPerPage: number
}

export const hpcApplicationsSearchQuery = ({query, page, itemsPerPage}: HPCApplicationsSearchQuery): string =>
    `/hpc/apps/search?query=${encodeURIComponent(query)}&page=${page}&itemsPerPage=${itemsPerPage}`;

export const hpcApplicationsTagSearchQuery = ({query, page, itemsPerPage}: HPCApplicationsSearchQuery): string =>
    `/hpc/apps/searchTags?query=${encodeURIComponent(query)}&page=${page}&itemsPerPage=${itemsPerPage}`;

export const cancelJobQuery = `hpc/jobs`;

export const cancelJobDialog = ({jobId, onConfirm, jobCount = 1}: { jobCount?: number, jobId: string, onConfirm: () => void }): void =>
    addStandardDialog({
        title: `Cancel job${jobCount > 1 ? "s" : ""}?`,
        message: jobCount === 1 ? `Cancel job: ${jobId}?` : "Cancel jobs?",
        cancelText: "No",
        confirmText: `Cancel job${jobCount > 1 ? "s" : ""}`,
        onConfirm
    });

export const cancelJob = (cloud: Cloud, jobId: string): Promise<{ request: XMLHttpRequest, response: void }> =>
    cloud.delete(cancelJobQuery, {jobId});

interface FavoriteApplicationFromPage<T> {
    name: string
    version: string
    page: Page<{ metadata: ApplicationMetadata, favorite: boolean } & T>
    cloud: Cloud
}

/**
 * Favorites an application.
 * @param {Application} Application the application to be favorited
 * @param {Cloud} cloud The cloud instance for requests
 */
export async function favoriteApplicationFromPage<T>({name, version, page, cloud}: FavoriteApplicationFromPage<T>): Promise<Page<T>> {
    const a = page.items.find(it => it.metadata.name === name && it.metadata.version === version)!;
    try {
        await cloud.post(hpcFavoriteApp(name, version));
        a.favorite = !a.favorite;
    } catch (e) {
        snackbarStore.addFailure(errorMessageOrDefault(e, `An error ocurred favoriting ${name}`));
    }
    return page;
}


interface StringMap {
    [k: string]: string;
}

interface AllowedParameterKey {
    name: string;
    type: ParameterTypes;
}

interface ExtractParameters {
    nameToValue: StringMap;
    allowedParameterKeys: AllowedParameterKey[];
    siteVersion: number;
}

export const findKnownParameterValues = ({nameToValue, allowedParameterKeys, siteVersion}: ExtractParameters): StringMap => {
    const extractedParameters = {};
    if (siteVersion === 1) {
        allowedParameterKeys.forEach(({name, type}) => {
            if (nameToValue[name] !== undefined) {
                if (typeMatchesValue(type, nameToValue[name])) {
                    extractedParameters[name] = nameToValue[name];
                }
            }
        });
    }
    return extractedParameters;
};

export const isFileOrDirectoryParam = ({type}: { type: string }) => type === "input_file" || type === "input_directory";

const typeMatchesValue = (type: ParameterTypes, parameter: string): boolean => {
    switch (type) {
        case ParameterTypes.Boolean:
            return parameter === "Yes" || parameter === "No" || parameter === "";
        case ParameterTypes.Integer:
            return parseInt(parameter, 10) % 1 === 0;
        case ParameterTypes.FloatingPoint:
            return typeof parseFloat(parameter) === "number";
        case ParameterTypes.Text:
        case ParameterTypes.InputDirectory:
        case ParameterTypes.InputFile:
        case ParameterTypes.SharedFileSystem:
        case ParameterTypes.Peer:
            return typeof parameter === "string";

    }
};

interface ExtractedParameters {
    [key: string]: string | number | boolean |
        { source: string, destination: string; } |
        { fileSystemId: string } |
        { jobId: string }
}

export type ParameterValues = Map<string, React.RefObject<HTMLInputElement | HTMLSelectElement>>;

interface ExtractParametersFromMap {
    map: ParameterValues;
    appParameters: ApplicationParameter[];
    cloud: Cloud;
}

export function extractValuesFromWidgets({map, appParameters, cloud}: ExtractParametersFromMap): ExtractedParameters {
    const extracted: ExtractedParameters = {};
    map.forEach(({current}, key) => {
        const parameter = appParameters.find(it => it.name === key);
        console.log(parameter, current);
        if (!current) return;
        if (!current.value || !current.checkValidity()) return;
        if (!parameter) return;
        switch (parameter.type) {
            case ParameterTypes.InputDirectory:
            case ParameterTypes.InputFile:
                const expandedValue = expandHomeFolder(current.value, cloud.homeFolder);
                extracted[key] = {
                    source: expandedValue,
                    destination: removeTrailingSlash(expandedValue).split("/").pop()!
                };
                return;
            case ParameterTypes.Boolean:
                switch (current.value) {
                    case "Yes":
                        extracted[key] = true;
                        return;
                    case "No":
                        extracted[key] = false;
                        return;
                    default:
                        return;
                }
            case ParameterTypes.Integer:
                extracted[key] = parseInt(current.value, 10);
                return;
            case ParameterTypes.FloatingPoint:
                extracted[key] = parseFloat(current.value);
                return;
            case ParameterTypes.Text:
                extracted[key] = current.value;
                return;
            case ParameterTypes.SharedFileSystem:
                console.log(current.value);
                extracted[key] = {fileSystemId: current.value};
                return;
            case ParameterTypes.Peer:
                extracted[key] = {jobId: current.value};
                return;
        }
    });
    return extracted;
}

export const inCancelableState = (state: AppState) =>
    state === AppState.VALIDATED ||
    state === AppState.PREPARED ||
    state === AppState.SCHEDULED ||
    state === AppState.RUNNING;


export function validateOptionalFields(
    invocation: ApplicationInvocationDescription,
    parameters: ParameterValues
): boolean {
    const optionalErrors = [] as string[];
    const optionalParams = invocation.parameters.filter(it => it.optional && it.visible).map(it =>
        ({name: it.name, title: it.title})
    );
    optionalParams.forEach(it => {
        const param = parameters.get(it.name)!;
        if (!param.current!.checkValidity()) optionalErrors.push(it.title);
    });

    if (optionalErrors.length > 0) {
        snackbarStore.addFailure(
            `Invalid values for ${optionalErrors.slice(0, 3).join(", ")}
                    ${optionalErrors.length > 3 ? `and ${optionalErrors.length - 3} others` : ""}`,
            5000
        );
        return false;
    }

    return true;
}

export function checkForMissingParameters(
    parameters: ExtractedParameters,
    invocation: ApplicationInvocationDescription
): boolean {
    const requiredParams = invocation.parameters.filter(it => !it.optional);
    const missingParameters: string[] = [];
    requiredParams.forEach(rParam => {
        const parameterValue = parameters[rParam.name];
        // Number, string, boolean 
        if (!parameterValue) {
            missingParameters.push(rParam.title);
        } else if (rParam.type === ParameterTypes.InputDirectory || rParam.type === ParameterTypes.InputFile) {
            if (!parameterValue["source"]) {
                missingParameters.push(rParam.title);
            }
        } else if (rParam.type === ParameterTypes.SharedFileSystem) {
            if (!parameterValue["fileSystemId"]) {
                missingParameters.push(rParam.title);
            }
        }
    });

    // Check missing values for required input fields.
    if (missingParameters.length > 0) {
        snackbarStore.addFailure(
            `Missing values for ${missingParameters.slice(0, 3).join(", ")} 
                ${missingParameters.length > 3 ? `and ${missingParameters.length - 3} others.` : ``}`,
            5000
        );
        return false;
    }
    return true;
}
