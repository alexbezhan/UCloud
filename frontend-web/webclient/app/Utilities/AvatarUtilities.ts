export const saveAvatarQuery = "/avatar/update";
export const findAvatarQuery = "/avatar/find";
