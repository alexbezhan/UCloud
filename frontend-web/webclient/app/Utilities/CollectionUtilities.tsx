export function removeEntry<T>(arr: T[], index: number): T[] {
    return arr.slice(0, index).concat(arr.slice(index + 1));
}

export function setsDiffer<T>(s1: Set<T>, s2: Set<T>): boolean {
    if (s1.size !== s2.size) return true;
    let differ = false;
    s1.forEach(it => {
        if (!s2.has(it)) differ = true;
    });
    return differ;
}

type PrimitiveDataTypes = string | number | boolean;

/**
 * @param set
 * @param entry
 */
export function addEntryIfNotPresent(set: Set<PrimitiveDataTypes>, entry: PrimitiveDataTypes): boolean {
    const size = set.size;
    set.add(entry);
    return size !== set.size;
}

export function groupBy<T>(items: T[], keySelector: (t: T) => string): Record<string, T[]> {
    const result: Record<string, T[]> = {};
    items.forEach(item => {
        const key = keySelector(item);
        const newList = result[key] ?? [];
        newList.push(item);
        result[key] = newList;
    });
    return result;
}

export function associateBy<T>(items: T[], keySelector: (t: T) => string): Record<string, T> {
    const result: Record<string, T> = {};
    items.forEach(item => {
        const key = keySelector(item);
        result[key] = item;
    });
    return result;
}

export function takeLast<T>(items: T[], numberOfItems: number): T[] {
    return items.slice(Math.max(0, items.length - numberOfItems));
}
