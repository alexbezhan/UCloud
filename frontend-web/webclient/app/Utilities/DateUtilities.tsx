import {format} from "date-fns/esm";

// Could potentially cause issues with time if user is outside CEST
export const dateToString = (date: number) => format(date, "HH:mm dd/MM/yyyy");
