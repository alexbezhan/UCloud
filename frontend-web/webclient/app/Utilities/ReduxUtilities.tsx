import {createStore, combineReducers, Store, AnyAction, Action} from "redux";
import {composeWithDevTools} from 'redux-devtools-extension';
import {ReduxObject, initObject} from "DefaultObjects";
import {responsiveBP} from "ui-components/theme";
import {createResponsiveStateReducer} from "redux-responsive";
import {Cloud} from "Authentication/SDUCloudObject";
import {CONTEXT_SWITCH, USER_LOGOUT, USER_LOGIN} from "Navigation/Redux/HeaderReducer";

export function configureStore(initialObject: Partial<ReduxObject>, reducers, enhancers?): Store<ReduxObject, AnyAction> {
    const combinedReducers = combineReducers<ReduxObject, AnyAction>(reducers);
    const rootReducer = (state: ReduxObject, action: Action): ReduxObject => {
        if ([USER_LOGIN, USER_LOGOUT, CONTEXT_SWITCH].some(it => it === action.type)) {
            state = initObject(Cloud.homeFolder);
        }
        return combinedReducers(state, action);
    };
    return createStore<ReduxObject, AnyAction, {}, {}>(rootReducer, initialObject, composeWithDevTools(enhancers));
}

export const responsive = createResponsiveStateReducer(
    responsiveBP,
    {infinity: "xxl"}
);