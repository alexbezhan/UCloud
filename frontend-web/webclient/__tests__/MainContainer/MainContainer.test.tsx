import * as React from "react";
import {Provider} from "react-redux";
import {create} from "react-test-renderer";
import {ThemeProvider} from "styled-components";
import {MainContainer} from "../../app/MainContainer/MainContainer";
import theme from "../../app/ui-components/theme";
import {store} from "../../app/Utilities/ReduxUtilities";

describe("Main container", () => {
    it("Only main container", () =>
        expect(create(
            <Provider store={store}>
                <ThemeProvider theme={theme}>
                    <MainContainer main={<div />} />
                </ThemeProvider>
            </Provider>
        ).toJSON()).toMatchSnapshot());

    it("Main container with sidebar", () =>
        expect(create(
            <Provider store={store}>
                <ThemeProvider theme={theme}>
                    <MainContainer main={<div />} sidebar={<div />} />
                </ThemeProvider>
            </Provider>
        ).toJSON()).toMatchSnapshot());

    it("Main container with header", () =>
        expect(create(
            <Provider store={store}>
                <ThemeProvider theme={theme}>
                    <MainContainer main={<div />} header={<div />} />
                </ThemeProvider>
            </Provider>
        ).toJSON()).toMatchSnapshot());
});
