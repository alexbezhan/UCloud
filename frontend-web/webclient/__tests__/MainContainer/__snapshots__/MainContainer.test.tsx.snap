// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main container Main container with header 1`] = `
.c0 {
  cursor: auto;
  margin-left: 190px;
  padding-top: 48px;
  padding-bottom: 14px;
  padding-left: 14px;
  padding-right: 0;
  background-color: #fff;
}

.c2 {
  cursor: auto;
  padding-top: 96px;
  padding-right: 14px;
}

.c3 {
  cursor: auto;
}

.c1 {
  cursor: auto;
  padding-top: 14px;
  padding-bottom: 14px;
  padding-left: 204px;
  padding-right: 14px;
  width: 100%;
  height: 96px;
  background-color: #fff;
  position: absolute;
  top: 48px;
  left: 0;
  position: fixed;
  z-index: 50;
}

@media screen and (min-width:768px) and (max-width:1023px) {
  .c3 {
    display: none;
  }
}

@media screen and (min-width:1024px) and (max-width:1279px) {
  .c3 {
    display: none;
  }
}

@media screen and (min-width:1280px) {
  .c3 {
    display: none;
  }
}

<div
  className="c0"
  cursor="auto"
>
  <div
    className="c1"
    cursor="auto"
    height={96}
    width={1}
  >
    <div />
  </div>
  <div
    className="c2"
    cursor="auto"
  >
    <div
      className="c3"
      cursor="auto"
    />
    <div />
  </div>
</div>
`;

exports[`Main container Main container with sidebar 1`] = `
.c0 {
  cursor: auto;
  margin-left: 190px;
  padding-top: 48px;
  padding-bottom: 14px;
  padding-left: 14px;
  padding-right: 0;
  background-color: #fff;
}

.c3 {
  cursor: auto;
  padding-top: 14px;
  padding-right: 240px;
}

.c4 {
  cursor: auto;
}

.c1 {
  cursor: auto;
}

.c2 {
  cursor: auto;
  padding-top: 62px;
  padding-left: 14px;
  padding-right: 14px;
  width: 240px;
  height: 100%;
  position: absolute;
  top: 0;
  right: 0;
  overflow-y: auto;
  position: fixed;
  z-index: 45;
}

@media screen and (min-width:768px) and (max-width:1023px) {
  .c4 {
    display: none;
  }
}

@media screen and (min-width:1024px) and (max-width:1279px) {
  .c4 {
    display: none;
  }
}

@media screen and (min-width:1280px) {
  .c4 {
    display: none;
  }
}

@media screen and (max-width:511px) {
  .c1 {
    display: none;
  }
}

@media screen and (min-width:512px) and (max-width:639px) {
  .c1 {
    display: none;
  }
}

@media screen and (min-width:640px) and (max-width:767px) {
  .c1 {
    display: none;
  }
}

<div
  className="c0"
  cursor="auto"
>
  <div
    className="c1"
    cursor="auto"
  >
    <div
      className="c2"
      cursor="auto"
      height="100%"
      width={240}
    >
      <div />
    </div>
  </div>
  <div
    className="c3"
    cursor="auto"
  >
    <div
      className="c4"
      cursor="auto"
    >
      <div />
    </div>
    <div />
  </div>
</div>
`;

exports[`Main container Only main container 1`] = `
.c0 {
  cursor: auto;
  margin-left: 190px;
  padding-top: 48px;
  padding-bottom: 14px;
  padding-left: 14px;
  padding-right: 0;
  background-color: #fff;
}

.c1 {
  cursor: auto;
  padding-top: 14px;
  padding-right: 14px;
}

.c2 {
  cursor: auto;
}

@media screen and (min-width:768px) and (max-width:1023px) {
  .c2 {
    display: none;
  }
}

@media screen and (min-width:1024px) and (max-width:1279px) {
  .c2 {
    display: none;
  }
}

@media screen and (min-width:1280px) {
  .c2 {
    display: none;
  }
}

<div
  className="c0"
  cursor="auto"
>
  <div
    className="c1"
    cursor="auto"
  >
    <div
      className="c2"
      cursor="auto"
    />
    <div />
  </div>
</div>
`;
